mm2src/coins/eth/eth_swap_v2.rs

```diff
@@ -96,14 +125,358 @@ impl EthCoin {
                 .compat()
                 .await
             },
-            EthCoinType::Nft { .. } => Err(TransactionErr::ProtocolNotSupported(
+            EthCoinType::Nft { .. } => Err(TransactionErr::ProtocolNotSupported(ERRL!(
+                "NFT protocol is not supported for ETH and ERC20 Swaps"
+            ))),
+        }
+    }
+
+    pub(crate) async fn validate_taker_funding_impl(
+        &self,
+        args: ValidateTakerFundingArgs<'_, Self>,
+    ) -> ValidateSwapV2TxResult {
+        if let EthCoinType::Nft { .. } = self.coin_type {
+            return MmError::err(ValidateSwapV2TxError::ProtocolNotSupported(
                 "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
-            )),
+            ));
+        }
+        let taker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.taker_swap_v2_contract)
+            .ok_or_else(|| {
+                ValidateSwapV2TxError::Internal("Expected swap_v2_contracts to be Some, but found None".to_string())
+            })?;
+        validate_payment_args(args.taker_secret_hash, args.maker_secret_hash, &args.trading_amount)
+            .map_err(ValidateSwapV2TxError::Internal)?;
+        let taker_address = public_to_address(args.taker_pub);
+        let funding_time_lock: u32 = args
+            .funding_time_lock
+            .try_into()
+            .map_to_mm(|e: TryFromIntError| ValidateSwapV2TxError::Overflow(e.to_string()))?;
+        let payment_time_lock: u32 = args
+            .payment_time_lock
+            .try_into()
+            .map_to_mm(|e: TryFromIntError| ValidateSwapV2TxError::Overflow(e.to_string()))?;
+        let swap_id = self.etomic_swap_id(payment_time_lock, args.maker_secret_hash);
```

Comment:
Q: Would switching [etomic_swap_id](https://github.com/KomodoPlatform/komodo-defi-framework/blob/35e92394928825c337f246f8e19fbfab1f65c4a8/mm2src/coins/eth.rs#L819-L820) argument to `u64` cause any problem? If we make this change, it can prevent us from the integer overflows.

Reply:
> Q: Would switching [etomic_swap_id](https://github.com/KomodoPlatform/komodo-defi-framework/blob/35e92394928825c337f246f8e19fbfab1f65c4a8/mm2src/coins/eth.rs#L819-L820) argument to `u64` cause any problem? If we make this change, it can prevent us from the integer overflows.

hmmm we can add `etomic_swap_id_v2(&self, time_lock: u64, secret_hash: &[u8])` and use it in tpu v2. 
not sure its a good idea to change swap ids in legacy swaps.

Reply:
in eth/nft swap v2 added etomic_swap_id_v2  https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/2ee4ef1cb1dbf21393a43a1596dc95cc60ba3d63